From 0cfa6ad3a6f1098c92ee7d5a2d6d9217d774d05e Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Sun, 24 Jul 2016 11:23:58 +0200
Subject: [PATCH] Bring back the module binding

Not sure why it was removed here and it breaks support for some devices.

http://www.draisberghof.de/usb_modeswitch/bb/viewtopic.php?f=2&t=2560
---
 40-usb_modeswitch.rules | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/40-usb_modeswitch.rules b/40-usb_modeswitch.rules
index 94bd742..5486491 100644
--- a/40-usb_modeswitch.rules
+++ b/40-usb_modeswitch.rules
@@ -8,7 +8,16 @@ ACTION!="add|change", GOTO="modeswitch_rules_end"
 # transfer; checked against a list of known modems, or else no action
 KERNEL=="ttyUSB*", ATTRS{bNumConfigurations}=="*", PROGRAM="usb_modeswitch --symlink-name %p %s{idVendor} %s{idProduct} %E{PRODUCT}", SYMLINK+="%c"
 
-SUBSYSTEM!="usb", ACTION!="add",, GOTO="modeswitch_rules_end"
+SUBSYSTEM!="usb", GOTO="modeswitch_rules_end"
+
+# Adds the device ID to the "option" driver after a warm boot
+# in cases when the device is yet unknown to the driver; checked
+# against a list of known modems, or else no action
+ATTR{bInterfaceClass}=="ff", ATTR{bInterfaceNumber}=="00", ATTRS{bNumConfigurations}=="*", RUN+="usb_modeswitch --driver-bind %p %s{idVendor} %s{idProduct} %E{PRODUCT}"
+
+
+# Don't continue on "change" event, prevent trigger by changed configuration
+ACTION!="add", GOTO="modeswitch_rules_end"
 
 
 # Generic entry for most Huawei devices, excluding Android phones
@@ -62,6 +71,9 @@ ATTR{idVendor}=="0421", ATTR{idProduct}=="0632", RUN+="usb_modeswitch '%b/%k'"
 # Nokia CS-21M-02
 ATTR{idVendor}=="0421", ATTR{idProduct}=="0637", RUN+="usb_modeswitch '%b/%k'"
 
+# Logitech G920 Racing Wheel
+ATTR{idVendor}=="046d", ATTR{idProduct}=="c261", RUN+="usb_modeswitch '%b/%k'"
+
 # Philips (?), Vodafone MD950 (Wisue Technology)
 ATTR{idVendor}=="0471", ATTR{idProduct}=="1210", RUN+="usb_modeswitch '%b/%k'"
 
-- 
2.7.4

